Automattic/wp-calypso
# state/utils: Add more types

#### Changes proposed in this Pull Request

* Adds types to `withEnhancers` ~and `withoutPersistence`~

#### Testing instructions



* Type checks pass. Only runtime change is to also pass `extraArguments` to `withEnhancers` which should have been there anyway.
